src/include/main/database.h

```diff
@@ -44,6 +44,7 @@ struct KUZU_API SystemConfig {
  */
 class Database {
     friend class EmbeddedShell;
+    friend class ClientContext;
```

Comment:
What's this change here for?

Reply:
This is to make ClientContext get the correct number of threads (i.e. the number from the system config).